src/main/java/user_code_resolver/UserCodeResolver.java

```diff
@@ -0,0 +1,222 @@
+                        String oldConstructorName = oldPathArray[oldPathArray.length - 1].substring(0, fileTypeStart);
+                        String newConstructorName = newPathArray[oldPathArray.length - 1].substring(0, fileTypeStart);
+                        String newPackage = oldPathArray[oldPathArray.length - 2];
+                        for (UserFiles file : project.getFiles()) {
+                            if (file.getFilename() == elementValueChange.getLeftValue()) {
+                                file.setFilename(newPath);
+                                for (int i = 0; i < file.getContent().size(); i++) {
+                                    String line = file.getContent().get(i);
+                                    if (line == null) continue;
+                                    if (line.contains("package")) {
+                                        for (int arrayIndex = oldPathArray.length - 3; arrayIndex >= 0; arrayIndex--) {
+                                            if (oldPathArray[arrayIndex].equals("java")) {
+                                                break;
+                                            } else {
+                                                newPackage = oldPathArray[arrayIndex] + "." + newPackage;
+                                            }
+                                        }
+                                        line = "package " + newPackage + ";";
+                                        file.getContent().set(i, line);
+                                    } else if (line.contains("class ") || line.contains(oldConstructorName)) {
+                                        file.getContent().set(i, line.replace(oldConstructorName, newConstructorName));
+                                    }
+                                }
+                            }
+
+                        }
+                    }
+                }
+            }
+        }
+        return project;
+    }
+
+
+    /**
+     * Read structure from folder as list
+     *
+     * @param folder the folder
+     * @return the list
+     * @throws IOException the io exception
+     */
+    private List<File> readStructureFromFolderAsList(final File folder) throws IOException {
+        List<File> filePaths = new ArrayList<>();
+        String path = String.valueOf(folder);
+        try (Stream<Path> paths = Files.walk(Paths.get(path))) {
+            paths.filter(Files::isRegularFile).map(Path::toFile).forEach(file -> {
+                if (file.toString().endsWith("Impl.java")) {
+                    filePaths.add(file);
+                }
+            });
+        }
+        return filePaths;
+    }
+
+    /**
+     * Read content of files and map as string.7
+     *
+     * @param fileList the file paths
+     * @return the string
+     */
+    private String readContentOfFilesAsString(List<File> fileList, Project project) throws JsonProcessingException {
+        BufferedReader reader;
```

Comment:
Kann hier weg, dafür in Zeile 173 dann `BufferedReader reader = new BufferedReader(new FileReader(userFile));`